### PR TITLE
Closes #2685 | Redesign menu & dialog – Main menu

### LINKED ISSUE #2685: Redesign menu & dialog – Main menu
Old:

New:

Spec:

Assets:

* [Number of trackers blocked](https://cl.ly/0R3e3N1N2G0j/trackers-blocked.svg)
* [Info](https://design.firefox.com/icons/viewer/#info)
* [Share…](https://design.firefox.com/icons/viewer/#share%20macos)
* [Home screen](https://design.firefox.com/icons/viewer/#home)
* [Open in…](https://design.firefox.com/icons/viewer/#open%20in%20new)
* [Request desktop site](https://design.firefox.com/icons/viewer/#desktop)
* [Settings](https://design.firefox.com/icons/viewer/#preferences)

Just a note that the icon fill follows our default value: `#f9f9fa` a80.

### PR DESCRIPTION
**Pull request by SrChip15**

### CODE REVIEW

**Comment by SrChip15** [comment]
@brampitoyo @ekager 

Screenshot
----

File: app/src/main/java/org/mozilla/focus/menu/browser/BrowserMenuAdapter.kt [hunk 1]
```diff
@@ -29,11 +29,11 @@ class BrowserMenuAdapter(
     sealed class MenuItem {
         open val viewType = 0
 
-        open class Default(val id: Int, val label: String) : MenuItem() {
+        open class Default(val id: Int, val label: String, val drawableResId: Int) : MenuItem() {
             override val viewType = MenuItemViewHolder.LAYOUT_ID
         }
 
-        class Custom(id: Int, label: String, val pendingIntent: PendingIntent) : Default(id, label) {
+        class Custom(id: Int, label: String, drawableResId: Int, val pendingIntent: PendingIntent) : Default(id, label, drawableResId) {
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/menu/browser/BrowserMenuAdapter.kt]: \[detekt\] INFO: \[style/MaxLineLength\] Line detected that is longer than the defined maximum line length in the code style\.

File: app/src/main/java/org/mozilla/focus/menu/browser/BrowserMenuAdapter.kt [hunk 2]
```diff
@@ -69,48 +69,48 @@ class BrowserMenuAdapter(
         items.add(MenuItem.BlockingSwitch)
 
         if (customTabConfig == null || customTabConfig.showShareMenuItem) {
-            items.add(MenuItem.Default(R.id.share, resources.getString(R.string.menu_share)))
+            items.add(MenuItem.Default(R.id.share, resources.getString(R.string.menu_share), R.drawable.ic_share_menu_item))
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/menu/browser/BrowserMenuAdapter.kt]: \[detekt\] INFO: \[style/MaxLineLength\] Line detected that is longer than the defined maximum line length in the code style\.

File: app/src/main/java/org/mozilla/focus/menu/browser/BrowserMenuAdapter.kt [hunk 3]
```diff
@@ -69,48 +69,48 @@ class BrowserMenuAdapter(
...
         }
 
-        items.add(MenuItem.Default(R.id.add_to_homescreen, resources.getString(R.string.menu_add_to_home_screen)))
-        items.add(MenuItem.Default(R.id.find_in_page, resources.getString(R.string.find_in_page)))
+        items.add(MenuItem.Default(R.id.add_to_homescreen, resources.getString(R.string.menu_add_to_home_screen), R.drawable.ic_add_to_home_menu_item))
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/menu/browser/BrowserMenuAdapter.kt]: \[detekt\] INFO: \[style/MaxLineLength\] Line detected that is longer than the defined maximum line length in the code style\.

File: app/src/main/java/org/mozilla/focus/menu/browser/BrowserMenuAdapter.kt [hunk 4]
```diff
@@ -69,48 +69,48 @@ class BrowserMenuAdapter(
...
+                    context.packageManager)), 0))
         }
 
+        items.add(MenuItem.RequestDesktopCheck)
+
         if (customTabConfig == null) {
             // There’s no need for Settings in a custom tab.
             // The user can go to the browser app itself in order to do this.
-            items.add(MenuItem.Default(R.id.settings, resources.getString(R.string.menu_settings)))
+            items.add(MenuItem.Default(R.id.settings, resources.getString(R.string.menu_settings), R.drawable.ic_settings_menu_item))
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/menu/browser/BrowserMenuAdapter.kt]: \[detekt\] INFO: \[style/MaxLineLength\] Line detected that is longer than the defined maximum line length in the code style\.

File: app/src/main/java/org/mozilla/focus/menu/browser/MenuItemViewHolder.java [hunk 5]
```diff
@@ -21,10 +21,12 @@
         menuItemView = (TextView) itemView;
     }
 
+
     /* package-private */ void bind(BrowserMenuAdapter.MenuItem.Default menuItem) {
         menuItemView.setId(menuItem.getId());
         menuItemView.setText(menuItem.getLabel());
-
+        menuItemView.setCompoundDrawablesRelativeWithIntrinsicBounds(menuItem.getDrawableResId(), 0, 0, 0);
+
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/menu/browser/MenuItemViewHolder.java]: \[Checkstyle\] ERROR: Trailing whitespace

**Pull request merged**